### PR TITLE
adding data attribute on the buttons 0-9

### PR DESCRIPTION
**Pull request by yaaHosein**
let us work together

### CODE REVIEW

File: calculator.js [hunk 1]
```diff
@@ -4,6 +4,8 @@ let lastButtonClicked = null;
 
 let button = document.getElementById("result");
 
+document.querySelectorAll("[data-number]")
```
Comment (SMH110) [inline calculator.js]: Very good, 
this function returns an array of elements (HTML elements) 
now you need to loop over array and for each HTML element you need to register a click event

Comment (yaaHosein) [reply calculator.js]: Ok!

Comment (yaaHosein) [reply calculator.js]: I have pushed some new updates

File: calculator.js [hunk 2]
```diff
@@ -5,164 +5,173 @@ let lastButtonClicked = null;
...
-equalButton.addEventListener("click", function() {
-    evaluate();
-});
\ No newline at end of file
+
+numBtnElem.forEach(function(button) {
+            button.addEventListener('click', function() {
+                    (lastButtonClicked === "Operator") result.textContent = "";
+                    const currentValueOfResult = result.textContent;
+                    const newValue = currentValueOfResult + Number;
```
Comment (SMH110) [inline calculator.js]: Instead of using `Number` which I don't where is coming from 
you should get the value of the button from the dataset of the button

Comment (yaaHosein) [reply calculator.js]: In the same way I used Number to assign a value for lastButtonClicked variable I thought let me use it here also!!

File: calculator.js [hunk 3]
```diff
@@ -4,6 +4,45 @@ let lastButtonClicked = null;
 
 let button = document.getElementById("result");
 
+let numBtnElem = document.querySelectorAll("[data-number]")
+    // numBtnElem.forEach(function(button) {
+    // button.addEventListener('click', function() {
+    //         (lastButtonClicked === "Operator") result.textContent = "";
+    //         const currentValueOfResult = result.textContent;
+    //         const newValue = currentValueOfResult + numBtnElem.dataset.number;
```
Comment (SMH110) [inline calculator.js]: ``` const newValue = currentValueOfResult + numBtnElem.dataset.number;```
This should have worked for you, but you're using the wrong object

`numBtnElem` is a collection of HTML elements.
You're using `forEach` to loop over this collection and you're passing a function to the `forEach` function
this function is passed a single element which you're adding event listener to it on line 9
you should have used this element to access the dataset property

File: calculator.js [hunk 4]
```diff
@@ -4,6 +4,19 @@ let lastButtonClicked = null;
 
 let button = document.getElementById("result");
 
+let numBtnElem = document.querySelectorAll("[data-number]")
+
+numBtnElem.forEach(function(button) {
+    button.addEventListener('click', function() {
+        (lastButtonClicked === "Operator") result.textContent = "";
```
Comment (SMH110) [inline calculator.js]: Can you explain what this line doing?

Comment (yaaHosein) [reply calculator.js]: Firstly 
there is if  in the beginning of it it I do not know how it fly
this line is to clear the result div after clicking on an opertaor button 
thanks

Comment (yaaHosein) [reply calculator.js]: should I do add data attribute to the operator buttons .... etc (loop) ?

Comment (SMH110) [reply calculator.js]: Yes, you can refactor the code of the + - / and * buttons following the same approach.

**Review by SMH110 — APPROVED** [review]

**Comment by SMH110** [comment]
very good work,
thank you

**Comment by yaaHosein** [comment]
All the continued thanks is for you

**Pull request merged**